CompilerSource/general/bettersystem.cpp

```diff
@@ -252,20 +256,90 @@ void myReplace(std::string& str, const std::string& oldStr, const std::string& n
       return e_exec(cmd, eCenviron);
     }
 #else
+    #include <vector>
+    #include <algorithm>
+    #include <cstdlib>
+    #include <csignal>
+
     #include <fcntl.h>
     #include <unistd.h>
     #include <sys/wait.h>
     #include <sys/stat.h>
+    #include <sys/types.h>
 
     extern char **environ;
     const mode_t laxpermissions = S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH;
 
-#if CURRENT_PLATFORM_ID ==  OS_MACOSX
+#if CURRENT_PLATFORM_ID == OS_MACOSX
+    #include <libproc.h>
+    #include <sys/proc_info.h>
     #include <crt_externs.h>
     #define environ (*_NSGetEnviron())
     extern char **environ;
 #endif
 
+    using std::vector;
+
+#if CURRENT_PLATFORM_ID == OS_FREEBSD
+    void ProcIdFromParentProcId(pid_t parentProcId, pid_t **procId, int *size) {
+      vector<pid_t> vec; int i = 0; int cntp;
```

Comment:
Why not just return the vector instead of copying it to a new address?

Reply:
Sorry, old habit. I prefer doing C abi usually for my work in case i want to port the function to a shared library, which std::vector is implementation defined and won't work across different compilers for that. Anyway, I fixed it.